File: src/main/java/com/m2t/hachikodictionary/dto/word/CreateWordRequest.java
package com.m2t.hachikodictionary.dto.word;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CreateWordRequest {
    @NotBlank
    private String title;
    @NotBlank
    private String kind;
    private String difficulty;
    @NotBlank
    private Set<String> description;
    private Set<String> synonyms;
    private Set<String> antonyms;
    private Set<String> sentences;
    private String fileUrl;
}

File: src/main/java/com/m2t/hachikodictionary/dto/CreateWordRequest.kt
package com.m2t.hachikodictionary.dto

import jakarta.validation.constraints.NotBlank

data class CreateWordRequest(
    @field:NotBlank(message = "Title cannot be blank.")
    val title: String,
    @field:NotBlank(message = "Kind cannot be blank.")
    val kind: String,
    val difficulty: String,
    @field:NotBlank(message = "Description cannot be blank.")
    val descriptions: Set<String>,
    val synonyms: Set<String>,
    val antonyms: Set<String>,
    val sentences: Set<String>,
) {

}